apps/tangle-dapp/data/blueprints/utils/toPrimitiveBlueprint.ts

```diff
@@ -0,0 +1,309 @@
+  TestFetcher,
+  WasmGadget,
+  WasmRuntime,
+} from '@webb-tools/tangle-substrate-types';
+
+export function toPrimitiveBlueprint({
+  metadata,
+  jobs,
+  registrationHook,
+  registrationParams,
+  requestHook,
+  requestParams,
+  gadget,
+}: ServiceBlueprint | TanglePrimitivesServicesServiceBlueprint) {
+  return {
+    metadata: toPrimitiveServiceMetadata(metadata),
+    jobs: jobs.map(toPrimitiveJobDefinition),
+    registrationHook: toPrimitiveServiceRegistrationHook(registrationHook),
+    requestHook: toPrimitiveServiceRequestHook(requestHook),
+    registrationParams: registrationParams.map(toPrimitiveFieldType),
+    requestParams: requestParams.map(toPrimitiveFieldType),
+    gadget: toPrimitiveGadget(gadget),
+  } as const;
+}
+
+export function toPrimitiveServiceMetadata({
+  name,
+  description,
+  author,
+  category,
+  codeRepository,
+  logo,
+  website,
+  license,
+}: ServiceMetadata) {
+  return {
+    name: u8aToString(name),
+    description: optionalBytesToString(description),
+    author: optionalBytesToString(author),
+    category: optionalBytesToString(category),
+    codeRepository: optionalBytesToString(codeRepository),
+    logo: optionalBytesToString(logo),
+    website: optionalBytesToString(website),
+    license: optionalBytesToString(license),
+  } as const;
+}
+
+function toPrimitiveJobDefinition({
+  metadata,
+  params,
+  result,
+  verifier,
+}: JobDefinition | TanglePrimitivesServicesJobDefinition) {
+  return {
+    metadata: toPrimitiveJobMetadata(metadata),
+    params: params.map(toPrimitiveFieldType),
+    result: result.map(toPrimitiveFieldType),
+    verifier: toPrimitiveJobResultVerifier(verifier),
+  } as const;
+}
+
+export function toPrimitiveServiceRegistrationHook(
+  registrationHook: ServiceRegistrationHook,
+) {
+  if (registrationHook.type === 'Evm') {
+    return {
+      Evm: registrationHook.asEvm.toHex(),
+    } as const;
+  }
+
+  return registrationHook.type;
+}
+
+export function toPrimitiveServiceRequestHook(requestHook: ServiceRequestHook) {
+  if (requestHook.type === 'Evm') {
+    return {
+      Evm: requestHook.asEvm.toHex(),
+    } as const;
+  }
+
+  return requestHook.type;
+}
+
+export function toPrimitiveJobResultVerifier(verifier: JobResultVerifier) {
+  if (verifier.type === 'Evm') {
+    return {
+      Evm: verifier.asEvm.toHex(),
+    } as const;
+  }
+
+  return verifier.type;
+}
+
+export function toPrimitiveJobMetadata({ name, description }: JobMetadata) {
+  return {
+    name: u8aToString(name),
+    description: optionalBytesToString(description),
+  } as const;
+}
+
+export function toPrimitiveGadget(gadget: Gadget) {
+  if (gadget.type === 'Container') {
+    return toPrimitiveContainerGadget(gadget.asContainer);
+  }
+
+  if (gadget.type === 'Native') {
+    return toPrimitiveNativeGadget(gadget.asNative);
+  }
+
+  if (gadget.type === 'Wasm') {
+    return toPrimitiveWasmGadget(gadget.asWasm);
+  }
+
+  throw new Error('Unknown Gadget type');
+}
```

Comment:
`switch` might be better suited here? Especially if `Container`, `Native` and `Wasm` are all the currently available possible value of `gadget.type`.